core/src/main/java/io/grpc/ServerServiceDefinition.java

```diff
@@ -121,7 +121,8 @@ private Builder(ServiceDescriptor serviceDescriptor) {
       MethodDescriptor<ReqT, RespT> method = def.getMethodDescriptor();
       checkArgument(
           serviceName.equals(MethodDescriptor.extractFullServiceName(method.getFullMethodName())),
-          "Service name mismatch. Expected service name: '%s'. Actual method name: '%s'.",
+          "Method name should be prefixed with service name and separated with '/'. "
+                  + "\n\tExpected service name: '%s'. Actual fully qualifed method name: '%s'.",
```

Comment:
I would not include tabs or new lines in the error message, because log parsers may not expect them.